### PR TITLE
Updating the lambda-trigger mdx with an example

### PR DESCRIPTION
**Pull request by brandonwatson**
#### Description of changes:
Related to https://github.com/aws-amplify/docs/pull/8006 - Josef asked that I figure out how to just get the single mdx change file updated. The packageManager kept trying to get into package.json but I eventually figured that out.

Single mdx file change.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/storage/lambda-triggers/index.mdx [hunk 1]
```diff
@@ -76,3 +76,32 @@ export const handler: S3Handler = async (event) => {
...
 
 Now, when you deploy your backend, these functions will be invoked whenever an object is uploaded or deleted from the bucket.
+
+## More Advanced Triggers
+
+The example listed above demonstrates what is exposed directly in your `storage` definition. Specifically, the use of the `triggers` option when you use `defineStorage`. This method is for simple triggers that always execute on file uploads or file deletions. There are no additional modifications you can make to the triggers defined in this way.
+
+If you want the ability to do something more than simply handle the events `onUpload` and `onDelete` you will have to use `.addEventNotification` in your `backend.ts`. If you use this method, the `triggers` section in your `storage/resource.ts` file should be removed.
+
+Here is an example of how you can add a lambda trigger for an s3 object PUT event. This trigger will execute when a file that has been uploaded to the bucket defined in your `storage/resource.ts` has a matching prefix and suffix as that listed in the function input of `addEventNotification`.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/storage/lambda-triggers/index.mdx:86]:
```suggestion
Here is an example of how you can add a Lambda trigger for an S3 object PUT event. This trigger will execute when a file that has been uploaded to the bucket defined in your `storage/resource.ts` has a matching prefix and suffix as that listed in the function input of `addEventNotification`.
```

small nit for service names

File: src/pages/[platform]/build-a-backend/storage/lambda-triggers/index.mdx [hunk 2]
```diff
@@ -76,3 +76,32 @@ export const handler: S3Handler = async (event) => {
...
+import { yourLambda } from './functions/yourLambda/resource';
+
+backend.storage.resources.bucket.addEventNotification(
+	EventType.OBJECT_CREATED_PUT,
+	new LambdaDestination(backend.yourLambda.resources.lambda),
+	{
+		prefix: 'protected/uploads/',
+		suffix: '-uploadManifest.json',
+	}
+);
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/storage/lambda-triggers/index.mdx:104]:
```suggestion
import { EventType } from 'aws-cdk-lib/aws-s3';
import { LambdaDestination } from 'aws-cdk-lib/aws-s3-notifications';
import { defineBackend } from '@aws-amplify/backend';
import { storage } from './storage/resource';
import { yourLambda } from './functions/yourLambda/resource';

const backend = defineBackend({
  storage,
  yourLambda,
});

backend.storage.resources.bucket.addEventNotification(
	EventType.OBJECT_CREATED_PUT,
	new LambdaDestination(backend.yourLambda.resources.lambda),
	{
		prefix: 'protected/uploads/',
		suffix: '-uploadManifest.json',
	}
);
```

and a small adjustment to define the backend variable

Comment (brandonwatson) [reply src/pages/[platform]/build-a-backend/storage/lambda-triggers/index.mdx:104]: As this is my first one of these, am I supposed to comment, or those comments for whoever's actually going to ultimately accept the pull request? I am fine with these changes.

Comment (josefaidt) [reply src/pages/[platform]/build-a-backend/storage/lambda-triggers/index.mdx:104]: Hey @brandonwatson, you can commit the suggestion or apply it manually. Either works!

**Review by josefaidt — APPROVED** [review]

**Review by cwomack — APPROVED** [review]

**Review by rtpascual — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**